packages/panels/src/Resources/Pages/CreateRecord.php

```diff
@@ -172,8 +172,9 @@ public function createAnother(): void
     protected function handleRecordCreation(array $data): Model
     {
         $record = new ($this->getModel())($data);
+        $tenant = Filament::getTenant();
 
-        if ($tenant = Filament::getTenant()) {
+        if ($tenant && static::getResource()::shouldApplyTenantRelationship()) {
```

Comment:
maybe check `shouldApplyTenantRelationship` before `$tenant = Filament::getTenant();` to avoid little query if not needed?